eng/native/build-commons.sh

```diff
@@ -94,6 +94,12 @@ build_native()
         cmakeArgs="-DCMAKE_SYSTEM_VARIANT=MacCatalyst $cmakeArgs"
     fi
 
+    if [[ "$targetOS" == "Linux" && "$__DistroRid" != "linux-musl-"* ]]; then
+        # When building on Linux, use the LLD linker
+        cmakeArgs="-DCMAKE_EXE_LINKER_FLAGS_INIT=-fuse-ld=lld -DCMAKE_SHARED_LINKER_FLAGS_INIT=-fuse-ld=lld $cmakeArgs"
```

Comment:
We are currently using `configuretools.cmake` or even `toolchain.cmake` (for crossbuild) to specify tools we prefer for the entire build. This way it is easier to remember where all the tools are defined. Can we have this condition there in the .cmake?

Reply:
The problem is that it would have to be duplicated in toolchain.cmake and configuretools.cmake. That was why I have moved it to the script. The toolchain.cmake runs before configuretools.cmake and obviously just for crossbuild.

Reply:
See my last global comment on this PR for details why the toolchain.cmake needs to have the linker already defined.

And regarding where we configure tools, we select compiler in the bash scripts, so selecting a linker seems to be a similar thing.

Reply:
Yup, toolchain.cmake is called early on by cmake, and we have some duplicates with configuretools. We also detect compiler in toolchain.cmake for some platforms, but you are correct about init-compiler.sh which selects compiler for majority. Could this be moved to init-compiler.sh and export LDFLAGS (next to CC, CXX and SCAN_BUILD_COMMAND)?

Reply:
Makes sense, I'll give it a try.